src/api/component.js

```diff
@@ -283,37 +281,70 @@ export default class extends HTMLElement {
       }
     }
 
-    if (attributeChanged) {
+    // DEPRECATED
+    //
+    // static attributeChanged()
+    const { attributeChanged } = this.constructor;
+    if (typeof attributeChanged === 'function') {
       attributeChanged(this, { name, newValue, oldValue });
     }
   }
 
   // Skate
   //
-  // Maps to the static updated() callback. That logic should be moved here
-  // when that is finally removed.
   updatedCallback (prev) {
-    return this.constructor.updated(this, prev);
+    // DEPRECATED
+    //
+    // static updated()
+    const { updated } = this.constructor;
+    if (typeof updated === 'function') {
+      return updated(this, prev);
+    }
+
+    // updatedCallback() default implementation
+    //
+    if (!prev) {
+      return true;
+    }
+
+    // use get all keys so that we check Symbols as well as regular props
+    // using a for loop so we can break early
+    const allKeys = getAllKeys(prev);
+    for (let i = 0; i < allKeys.length; i += 1) {
```

Comment:
`for` is error prone and hard to read, rather use `for .. of`, much smaller footprint and ceremony

```js
  for(const key of allKeys){
    if(prev[key] !== this[key]){
      return true;
    }
  }
```